main.go

```diff
@@ -185,3 +223,77 @@ func setupWebhooks(mgr ctrl.Manager) {
 func concurrency(c int) controller.Options {
 	return controller.Options{MaxConcurrentReconciles: c}
 }
+
+// GetTLSOptionOverrideFuncs returns a list of TLS configuration overrides to be used
+// by the webhook server.
+func GetTLSOptionOverrideFuncs(options TLSOptions) ([]func(*tls.Config), error) {
+	var tlsOptions []func(config *tls.Config)
+	var TLSMinVersionIndex, TLSMaxVersionIndex int
+	for tlsVersionIndex, TLSVersion := range cliflag.TLSPossibleVersions() {
+		if options.TLSMinVersion == TLSVersion {
+			TLSMinVersionIndex = tlsVersionIndex
+		}
+		if options.TLSMaxVersion == TLSVersion {
+			TLSMaxVersionIndex = tlsVersionIndex
+		}
+	}
+	// Check if tls min version flag value is greater than max version
+	if TLSMinVersionIndex > TLSMaxVersionIndex &&
+		TLSMaxVersionIndex != -1 &&
+		TLSMinVersionIndex != -1 {
+		return nil, fmt.Errorf("TLS version flag min version is greater than max version")
+	}
+
+	tlsMinVersion, err := cliflag.TLSVersion(options.TLSMinVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MinVersion = tlsMinVersion
+	})
+
+	tlsMaxVersion, err := cliflag.TLSVersion(options.TLSMaxVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MaxVersion = tlsMaxVersion
+	})
+	// Cipher suites should not be set if empty.
+	if options.TLSMinVersion == "VersionTLS13" && options.TLSMaxVersion == "VersionTLS13" {
+		setupLog.Info("warning: cipher suite flag set to null for TLS version 1.3")
+		options.TLSCipherSuites = ""
+	}
+
+	if options.TLSCipherSuites != "" {
+		tlsCipherSuites := strings.Split(options.TLSCipherSuites, ",")
+		suites, err := cliflag.TLSCipherSuites(tlsCipherSuites)
+		if err != nil {
+			return nil, err
+		}
+
+		insecureCipherValues := cliflag.InsecureTLSCipherNames()
+		for _, cipher := range tlsCipherSuites {
+			for _, insecureCipherName := range insecureCipherValues {
+				if insecureCipherName == cipher {
+					setupLog.Info(fmt.Sprintf("warning: use of insecure cipher '%s' detected.", cipher))
+				}
+			}
+		}
+		tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+			cfg.CipherSuites = suites
+		})
+	}
+
+	if options.TLSCertFile != "" && options.TLSPrivateKeyFile != "" {
```

Comment:
If only 1 option is specified we should error out I think. Now we just silently ignore it

Reply:
@lentzi90 are these required parameters? Because I assumed these were optional values and designed test cases around that scenario. If these are required. What should be their values i.e. is there some place where we have these files located? Otherwise if we provide dummy values for these it will also error out in that scenario.

Reply:
No not required. But if a cert is specified and no key, then that is an error. I'm open to suggestions on how to handle it but I don't think it is ok that the user can set `--tls-cert-file=/my/certificate` and it is silently ignored.

Example:

- `--tls-cert-file` is specified but not `--tls-private-key-file` -> error
- `--tls-private-key-file` is specified but not`--tls-cert-file` -> error
- both flags are specified -> success
- no flags are specified -> success (don't use TLS)

Reply:
> * no flags are specified -> success (don't use TLS)

This is not something we want to support. Currently when we don't provide any certs or keys, we use cert-manager provided certs. So if no flags are specified, we keep using default certs.

Reply:
Right. Let's just get rid of these flags. The webhook-cert-dir is enough and it just makes it confusing/complex to have separate flags for cert and key.